### PR TITLE
VBID-10250: Enable fallback assets

### PR DESCRIPTION
**Pull request by dmansen**
Add fallback asset support. This reads from a directory,
config.fallbackAssetPath. This directory can contain any number of
images.

If we have no ads to show, use images from the fallback directory.
These will show with a duration of 1s, the same as our current retry
rate for failures.

### CODE REVIEW

**Comment by dmansen** [comment]
Still working on getting the player set up for real testing, so unknown if this actually works on the real player so far.

File: src/fallback_assets.coffee [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  ads: []
+
+  constructor: ->
+    @refresh()
+
+  refresh: ->
+    if @_config.fallbackAssetPath
+      files = @_fs.readdirSync(@_config.fallbackAssetPath)
+      @ads = for f in files
+        asset_url: "file://#{f}"
```
Comment (psimmons-vistar) [inline src/fallback_assets.coffee]: should we try to derive the dimensions at all so we can know which asset will fit to the screen it is playing on, or is Product cool with that being configured by media owners?

**Comment by psimmons-vistar** [comment]
By real player do you mean the Ayuda player on top of the Ayuda asset, or the bear html5player?

File: src/fallback_assets.coffee (hunk 1 above)
Comment (dmansen) [reply src/fallback_assets.coffee]: I'll ask Christian. He seems to want the easiest possible solution so I doubt it

**Comment by dmansen** [comment]
As far as I can tell, our ayuda player JS project runs on the Ayuda "Splash Player" which I have installed on a VM but can't seem to figure out yet. So it's the combination of those 3, splash player + ayuda player + html5player

**Comment by dmansen** [comment]
From reading Ayuda's docs in more detail, it seems we need to use their "Forward and Store" feature referenced in the ticket. Then, when we reference the magic directory in a URL, it'll allow us to access the filesystem. I'll update.

**Comment by dmansen** [comment]
Alright. Chris was right and I forgot this is a full HTML app, so definitely no filesystem access. They'll need to store the exact path to the asset in the config and we'll show it if it's there. Only one asset allowed, no size requirements (ran these by Christian).

**Comment by dmansen** [comment]
Simplified the code a lot - we only have a single asset now, and aren't reading from the filesystem. Tested successfully with the bare html5player.

Working with platform to test this on the ayuda player, but it might not be possible due to the way things are set up on the media owner's end.

File: src/varied_ad_stream.coffee [hunk 2]
```diff
@@ -53,20 +55,37 @@ class VariedAdStream extends VarietyStream
             callback(ads)
           .catch ->
             succeeded = (ad for ad, i in ads when not downloads[i].failed)
```
Comment (psimmons-vistar) [inline src/varied_ad_stream.coffee]: nitty might be worth renaming this value for succeeded, maybe downloaded or available

**Comment by dmansen** [comment]
uploading a new request, this was mostly rewritten

**Pull request closed without merging**